deeplake/core/vectorstore/deeplake_vectorstore.py

```diff
@@ -286,6 +322,16 @@ def add(
             logger=logger,
         )
 
+        if self.exec_option in ('tensor_db', 'compute_engine'):
```

Comment:
@sounakr Since this logic is applied at least twice in different places, pls break it out into a separate utility function so it's easier to update in the future.